Bearded.Utilities/Noise/INoiseMap.cs

```diff
@@ -0,0 +1,41 @@
+namespace Bearded.Utilities.Noise
+{
+    public interface INoiseMap
```

Comment:
Renamed to `ProceduralTexture` since according to [Wikipedia](https://en.wikipedia.org/wiki/Procedural_texture), that is the right name for something like this. Perlin noise specifically is mentioned as an example. I'm open to renaming the entire namespace, or move the non-noise specific procedural texture types to `Core` or their own namespace.